shopelectro/tests/tests_models.py

```diff
@@ -86,3 +86,32 @@ def test_get_brands_from_products(self):
                 p: p.tags.filter(group__name=settings.BRAND_TAG_GROUP_NAME).first()
                 for p in products
             }
+
+    def test_get_catalog_blocks(self):
+        """Perform four queries to fetch in batch blocks, categories, pages and category's children."""
+        roots = Category.objects.filter(level=0)
+        roots_count = roots.count()
+        for category in roots:
+            CatalogBlock.objects.create(category=category)
+
+        with self.assertNumQueries(4):
```

Comment:
This structure would be more clear. Up to you

```
with self.assertNumQueries(1):
    # fetch blocks
with self.assertNumQueries(3):
    # fetch category, page, rows
```

Reply:
Looks better, but `blocks` can't be fetched in 1 query, because of `select_related('category', 'cateogyr__page')` and `rows` takes 2 queries, so this structure be more relevant:
``` 
with self.assertNumQueries(2):
    # fetch blocks, category, page
with self.assertNumQueries(2):
    # fetch rows
```
P.S.: [To make clear](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#prefetch-related) the point about `select_related`
>select_related gets the related objects in the same database query